generators/contract/templates/java/src/main/java/org/example/MyContract.java

```diff
@@ -0,0 +1,90 @@
+                version = "<%= version %>", 
+                license = 
+                        @License(name = "<%= spdxAndLicense // SPDX-License-Identifier: Apache-2.0 %>", 
+                                url = ""), 
+                                contact =  @Contact(email = "<%= name %>@example.com", 
+                                                name = "<%= name %>", 
+                                                url = "http://<%= name %>.me")))
+@Default
+public class <%= assetPascalCase %>Contract implements ContractInterface {
+    public  <%= assetPascalCase %>Contract() {
+
+    }
+    @Transaction()
+    public boolean <%= assetCamelCase %>Exists(Context ctx, String <%= assetCamelCase %>Id) {
+        byte[] buffer = ctx.getStub().getState(<%= assetCamelCase %>Id);
+        return (buffer != null && buffer.length > 0);
+    }
+
+    @Transaction()
+    public void create<%= assetPascalCase %>(Context ctx, String <%= assetCamelCase %>Id, String value) {
+        boolean exists = <%= assetCamelCase %>Exists(ctx,<%= assetCamelCase %>Id);
+        if (exists) {
+            throw new RuntimeException("The asset "+<%= assetCamelCase %>Id+" already exists");
+        }
+        <%= assetPascalCase %> asset = new <%= assetPascalCase %>();
+        asset.setValue(value);
+        ctx.getStub().putState(<%= assetCamelCase %>Id, asset.toJSONString().getBytes(UTF_8));
+    }
+
+    @Transaction()
+    public <%= assetPascalCase %> read<%= assetPascalCase %>(Context ctx, String <%= assetCamelCase %>Id) {
+        boolean exists = <%= assetCamelCase %>Exists(ctx,<%= assetCamelCase %>Id);
+        if (!exists) {
+            throw new RuntimeException("The asset "+<%= assetCamelCase %>Id+" does not exist");
+        }
+
+        <%= assetPascalCase %> newAsset = <%= assetPascalCase %>.fromJSONString(new String(ctx.getStub().getState(<%= assetCamelCase %>Id),UTF_8));
+        return newAsset;
+    }
+
+    @Transaction()
+    public void update<%= assetPascalCase %>(Context ctx, String <%= assetCamelCase %>Id, String newValue) {
+        boolean exists = <%= assetCamelCase %>Exists(ctx,<%= assetCamelCase %>Id);
+        if (!exists) {
+            throw new RuntimeException("The asset "+<%= assetCamelCase %>Id+" does not exist");
+        }
+        <%= assetPascalCase %> asset = new <%= assetPascalCase %>();
+        asset.setValue(newValue);
+        
+        ctx.getStub().putState(<%= assetCamelCase %>Id, asset.toJSONString().getBytes(UTF_8));
+    }
+
+    @Transaction()
+    public void delete<%= assetPascalCase %>(Context ctx, String <%= assetCamelCase %>Id) {
+        boolean exists = <%= assetCamelCase %>Exists(ctx,<%= assetCamelCase %>Id);
+        if (!exists) {
+            throw new RuntimeException("The asset "+<%= assetCamelCase %>Id+" does not exist");
+        }
+        ctx.getStub().delState(<%= assetCamelCase %>Id);
+    }
+    
+    @Override
+    public void afterTransaction(Context ctx, Object value) {
```

Comment:
Do we need these two functions? They're not in the JS/TS versions?

Reply:
No they can go.. not being used here